### DISCUSSION TITLE
Bluetooth connection hangs / too many messages - with v3.3.0b6

### DISCUSSION THREAD

**Discussion by Debenben** [pybricks/support#1091] [Q&A]
I can reproduce something similar to the reported problem with v3.3.0b6 and the following code:

```python
from pybricks.hubs import TechnicHub
from pybricks.parameters import Color

hub = TechnicHub(observe_channels=[0], broadcast_channel=1)

while True:
    hub.ble.observe(0)
    hub.light.on(Color.RED)
    hub.ble.broadcast("hello")
    print("running")
```
When running the above program the hub stops broadcasting "hello" after around 10 to 40 seconds. If connected to a computer it also stops printing "running" in the pybricks-ide. When connected to the computer and using long-press the hub starts blinking blue very fast and is no longer responsive, I need to remove the batteries to get it back to normal. If not connected to the computer I can stop the program with long-press on the button and then the hub behavior is back to normal. 

The hub.light.on and print command is not necessary, the behavior is also there when only using hub.ble.observe and hub.ble.broadcast commands. The crash and weired behavior is gone when adding wait(1) in the loop.

**Comment by dlech** [comment]
I moved this to an issue since it is a bug report. https://github.com/pybricks/support/issues/1095
